### PR TITLE
Add memory backend

### PR DESCRIPTION
**Pull request by leplatrem**
Revamp of #78 with just MemoryBackend.

After discussion with @AntoineCezar, it makes sense to get rid of couchdb backend for testing.
However, we should have specific tests for each backend.

### CODE REVIEW

File: daybed/validators.py [hunk 1]
```diff
@@ -35,11 +36,34 @@ def schema_validator(request):
 
 
 def validate_against_schema(request, schema, data):
+    """Validates data against specified schema, and attach
+    a deserialized and cleaned-up version of data to the request.
+    """
     try:
-        request.data_clean = schema.deserialize(data)
```
Comment (almet) [inline daybed/validators.py]: better naming, thanks.

**Comment by almet** [comment]
We need to implement tests for the CouchDB backend before we can merge this.

Also, can we have tests for the new MemoryBackend?

**Comment by AntoineCezar** [comment]
I'll write them, if not assigned.

File: daybed/validators.py (hunk 1 above)
Comment (leplatrem) [inline daybed/validators.py]: was that a recommandation ? 'cuz `data_clean` attribute was not renamed in this commit...

Comment (almet) [inline daybed/validators.py]: sorry if that was misleading, I'm just saying that using data_pure and data_clean seems better to me than the previous version

**Comment by almet** [comment]
We've been adding the tests in the ACL branch :-)

**Comment by almet** [comment]
We now have proper tests for the backends, feel freee to update this code to the new interfaces exposed and do another pull request :)

**Comment by leplatrem** [comment]
Unfortunately backend tests were done in ACL branch :)

I suggest we wait for ACL to be merged in master. And then, just add a `BackendTestCase` to this same PR and merge it !

**Comment by almet** [comment]
oh, right. +1 to this. @Natim do you have time to finish the work on ACL or should I take a stab at it?

**Comment by Natim** [comment]
Now that it has been merged :) We can refactor to test both CouchDB Backend and memory backend.

**Comment by leplatrem** [comment]
I'll take this one, before the end of next week :)

**Comment by almet** [comment]
Really cool. When this will be done, please switch the tests to this backend by default so we avoid spending so much time during network calls. That should speedup the tests.

**Pull request merged**